atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraAtlasDbFactory.java

```diff
@@ -42,36 +42,32 @@
 import com.palantir.util.OptionalResolver;
 import java.util.Optional;
 import java.util.function.LongSupplier;
-import java.util.function.Supplier;
 
 @AutoService(AtlasDbFactory.class)
-public class CassandraAtlasDbFactory implements AtlasDbFactory<CassandraReloadableKvsConfig> {
+public class CassandraAtlasDbFactory implements AtlasDbFactory<CassandraReloadableKeyValueServiceRuntimeConfig> {
     private static SafeLogger log = SafeLoggerFactory.get(CassandraAtlasDbFactory.class);
-    private CassandraKeyValueServiceRuntimeConfig latestValidRuntimeConfig =
-            CassandraKeyValueServiceRuntimeConfig.getDefault();
 
     @Override
     public KeyValueService createRawKeyValueService(
             MetricsManager metricsManager,
-            CassandraReloadableKvsConfig config,
-            Refreshable<Optional<KeyValueServiceRuntimeConfig>> runtimeConfig,
+            KeyValueServiceConfig config,
+            Refreshable<Optional<CassandraReloadableKeyValueServiceRuntimeConfig>> runtimeConfig,
             Optional<LeaderConfig> unused,
             Optional<String> namespace,
             LongSupplier freshTimestampSource,
             boolean initializeAsync) {
         AtlasDbVersion.ensureVersionReported();
-        Supplier<CassandraKeyValueServiceRuntimeConfig> cassandraRuntimeConfig =
-                preprocessKvsRuntimeConfig(runtimeConfig);
+
         return CassandraKeyValueServiceImpl.create(
                 metricsManager,
                 toCassandraConfig(config),
-                cassandraRuntimeConfig,
+                preprocessKvsRuntimeConfig(runtimeConfig),
                 CassandraMutationTimestampProviders.singleLongSupplierBacked(freshTimestampSource),
                 initializeAsync);
     }
 
     @Override
-    public CassandraReloadableKvsConfig createMergedKeyValueServiceConfig(
+    public Refreshable<Optional<CassandraReloadableKeyValueServiceRuntimeConfig>> createMergedKeyValueServiceConfig(
```

Comment:
I'm skeptical about returning this type; why aren't we able to just return `CassandraKeyValueServiceRuntimeConfig`?

Reply:
Discussed - all of the relevant methods take in a `Refreshable<Optional<KeyValueServiceRuntimeConfig>>` (and `AtlasDbRuntimeConfig#keyValueService()` returns an `Optional<KeyValueServiceRuntimeConfig>`. However, I was able to remove this method entirely and keep the merged config as an implementation detail in `getRawKeyValueService`